File: app/src/main/java/com/ndmsystems/coala/message/CoAPMessagePayload.java
package com.ndmsystems.coala.message;


public class CoAPMessagePayload {
    public byte[] content;

    public CoAPMessagePayload(byte[] content) {
        this.content = content;
    }

    public CoAPMessagePayload(String stringContent) {
        content = stringContent.getBytes();
    }

    public String toString() {
        if (content == null) return null;
        else return new String(content);
    }
}

File: app/src/main/java/com/ndmsystems/coala/message/CoAPMessagePayload.kt
package com.ndmsystems.coala.message

import java.nio.charset.StandardCharsets

class CoAPMessagePayload {
    @JvmField
    var content: ByteArray

    constructor(content: ByteArray) {
        this.content = content
    }

    constructor(stringContent: String) {
        content = stringContent.toByteArray(StandardCharsets.UTF_8)
    }

    override fun toString(): String {
        return if (content.isEmpty()) "" else String(content, StandardCharsets.UTF_8)
    }
}
